rust/agama-lib/src/storage/model/dasd.rs

```diff
@@ -0,0 +1,39 @@
+//! Implements a data model for DASD devices management.
+use std::collections::HashMap;
+
+use serde::Serialize;
+use zbus::zvariant::OwnedValue;
+
+use crate::{dbus::get_property, error::ServiceError};
+
+/// Represents a DASD device (specific to s390x systems).
+#[derive(Clone, Debug, Serialize, Default, utoipa::ToSchema)]
+pub struct DASDDevice {
```

Comment:
Some of the fields in this struct (`status`, `device_type` and/or `access_type`) might deserve an enum.